cloud/db/organization-memberships.test.ts

```diff
@@ -0,0 +1,1304 @@
+      () =>
+        Effect.gen(function* () {
+          const { org, owner, nonMember } =
+            yield* TestEffectOrganizationFixture;
+          const db = yield* EffectDatabase;
+
+          // First add succeeds
+          yield* db.organizations.memberships.create({
+            userId: owner.id,
+            organizationId: org.id,
+            data: { memberId: nonMember.id, role: "MEMBER" },
+          });
+
+          // Second add fails
+          const result = yield* db.organizations.memberships
+            .create({
+              userId: owner.id,
+              organizationId: org.id,
+              data: { memberId: nonMember.id, role: "ADMIN" },
+            })
+            .pipe(Effect.flip);
+
+          expect(result).toBeInstanceOf(AlreadyExistsError);
+        }),
+    );
+
+    it.effect("returns `DatabaseError` when insert fails", () =>
+      Effect.gen(function* () {
+        const db = yield* EffectDatabase;
+
+        const result = yield* db.organizations.memberships
+          .create({
+            userId: "owner-id",
+            organizationId: "org-id",
+            data: { memberId: "target-id", role: "MEMBER" },
+          })
+          .pipe(Effect.flip);
+
+        expect(result).toBeInstanceOf(DatabaseError);
+        expect(result.message).toBe("Failed to create organization membership");
+      }).pipe(
+        Effect.provide(
+          new MockDrizzleORM()
+            // authorize: getRole -> getMembership returns OWNER
+            .select([{ role: "OWNER" }])
+            // insert membership: fails
+            .insert(new Error("Connection failed"))
+            .build(),
+        ),
+      ),
+    );
+
+    it.effect("returns `DatabaseError` when audit log insert fails", () =>
+      Effect.gen(function* () {
+        const db = yield* EffectDatabase;
+
+        const result = yield* db.organizations.memberships
+          .create({
+            userId: "owner-id",
+            organizationId: "org-id",
+            data: { memberId: "target-id", role: "MEMBER" },
+          })
+          .pipe(Effect.flip);
+
+        expect(result).toBeInstanceOf(DatabaseError);
+        expect(result.message).toBe("Failed to create audit log");
+      }).pipe(
+        Effect.provide(
+          new MockDrizzleORM()
+            // authorize: getRole -> getMembership returns OWNER
+            .select([{ role: "OWNER" }])
+            // insert membership: succeeds
+            .insert([
+              { memberId: "target-id", role: "MEMBER", createdAt: new Date() },
+            ])
+            // insert audit log: fails
+            .insert(new Error("Connection failed"))
+            .build(),
+        ),
+      ),
+    );
+  });
+
+  // ===========================================================================
+  // Find All
+  // ===========================================================================
+
+  describe("findAll", () => {
```

Comment:
Let's add a case where user was added to org, deleted, and check the findAll semantics.